apps/website/src/server/db/short-links.ts

```diff
@@ -0,0 +1,54 @@
+import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import { prisma } from "@/server/db/client";
+import { SLUG_REGEX } from "@/utils/slugs";
+
+export const shortLinkSchema = z.object({
+  label: z.string().min(1),
+  slug: z.string().regex(SLUG_REGEX),
+  link: z.string().url(),
+});
+
+export type ShortLinkSchema = z.infer<typeof shortLinkSchema>;
+
+export const existingShortLinkSchema = shortLinkSchema.and(
+  z.object({
+    id: z.string().cuid(),
+  }),
+);
+
+export async function createShortLink(input: z.infer<typeof shortLinkSchema>) {
+  const slug = input.slug;
+  const existingShortLinkWithSlug = await prisma.shortLinks.findFirst({
+    where: { slug },
+  });
+  if (existingShortLinkWithSlug)
+    throw new TRPCError({
+      code: "BAD_REQUEST",
+      message: "Slug already exists",
+    });
+
+  return await prisma.shortLinks.create({
+    data: { ...input, slug },
+  });
```

Comment:
```suggestion
  const existingShortLinkWithSlug = await prisma.shortLinks.findFirst({
    where: { slug: input.slug },
  });
  if (existingShortLinkWithSlug)
    throw new TRPCError({
      code: "BAD_REQUEST",
      message: "Slug already exists",
    });

  return await prisma.shortLinks.create({
    data: input,
  });
```
We can clean this up a bit now we don't have the fallback